pkg/cleaner/cleaner.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cleaner
+
+import (
+	"fmt"
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	"github.com/kubernetes-incubator/service-catalog/pkg/probe"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"time"
+)
+
+type Cleaner struct {
+	client              kubernetes.Interface
+	scClient            sc.Interface
+	apiextensionsClient apiextensionsclientset.Interface
+}
+
+// New returns new Cleaner struct
+func New(
+	k8sclient kubernetes.Interface,
+	serviceCatalogClient sc.Interface,
+	apiExtClient apiextensionsclientset.Interface) *Cleaner {
+	return &Cleaner{
+		client:              k8sclient,
+		scClient:            serviceCatalogClient,
+		apiextensionsClient: apiExtClient,
+	}
+}
+
+// RemoveCRDs takes four steps, first scale down controller manager deployment, second removes all ServiceCatalog CRDs,
+// third removes all finalizers from CRs and the last step makes sure all ServiceCatalog CRDs are removed
+func (c *Cleaner) RemoveCRDs(releaseNamespace, controllerManagerName string) error {
+	err := c.scaleDownController(releaseNamespace, controllerManagerName)
+	if err != nil {
+		return fmt.Errorf("failed to scale down controller manager: %v", err)
+	}
+
+	err = c.removeCRDs(c.apiextensionsClient)
+	if err != nil {
+		return fmt.Errorf("failed to remove CustomResourceDefinitions: %v", err)
+	}
+
+	klog.V(4).Info("Removing finalizers from all ServiceCatalog custom resources")
+	finalizerCleaner := NewFinalizerCleaner(c.scClient)
+	err = finalizerCleaner.RemoveFinalizers()
+	if err != nil {
+		return fmt.Errorf("failed to remove finalizers from ServiceCatalog CRs: %s", err)
+	}
+
+	err = c.checkCRDsNotExist(c.apiextensionsClient)
+	if err != nil {
+		return fmt.Errorf("failed while checking CRDs not exist: %s", err)
+	}
+
+	return nil
+}
+
+func (c *Cleaner) scaleDownController(namespace, controllerName string) error {
+	klog.V(4).Infof("Fetching deployment %s/%s", namespace, controllerName)
+	deployment, err := c.client.
```

Comment:
please inline that. It will be consistent with rest of your code, e.g. line 91, 98 etc.